### DISCUSSION TITLE
walletconnect/modal-react-native Error "Invariant Violation: Tried to register two views with the same name RNSVGRect"

### DISCUSSION THREAD

**Discussion by on99yu** [WalletConnect/walletconnect-monorepo#3322] [Developer Support]
When i use package - "walletconnect/modal-react-native" in react native JS file 

This Error occured "Invariant Violation: Tried to register two views with the same name RNSVGRect"
I tried just to remove "node_modules" and npm install but it couldn't resolved
How can i resolve this

**Comment by ignaciosantise** [comment]
hey @on99yu 👋 can you share a public repo with the reproducible code? Also steps to reproduce it

**Reply by on99yu** [reply]
https://github.com/on99yu/nangnang
here is repo. project name is nangnang
Error occured in nangnang/screens/SelectWallet.js

**Comment by saber-yy** [comment]
check yarn.lock, and make sure there are only one react-native-svg version.

**Comment by ignaciosantise** [comment]
hey @on99yu there are some things you can do to solve the issue:

1. Remove `@walletconnect/react-native-dapp` as it's deprecated
2. Update `react-native-svg` to it's latest version -> `yarn add react-native-svg@13.10.0`

Also, i suggest you use either yarn or npm to install packages but not both. If you use yarn, remove the file `package-lock.json`

**Reply by on99yu** [reply]
Thank you so much 
But after that i remove "@walletconnect/react-native-dapp" 
How should i change the "index.js" file

`// registerRootComponent calls AppRegistry.registerComponent('main', () => App);
// It also ensures that whether you load the app in Expo Go or in a native build,
// the environment is set up appropriately
const { Platform } = require("react-native");

if (Platform.OS !== "web") {
require("./global");
}

const { registerRootComponent, scheme } = require("expo");
const { default: App } = require("./App");

const {
default: AsyncStorage,
} = require("@react-native-async-storage/async-storage");
const { withWalletConnect } = require("@walletconnect/react-native-dapp");

// registerRootComponent calls AppRegistry.registerComponent('main', () => App);
// It also ensures that whether you load the app in the Expo client or in a native build,
// the environment is set up appropriately
registerRootComponent(
withWalletConnect(App, {
redirectUrl:
Platform.OS === "web" ? window.location.origin :`${scheme}://`,
storageOptions: {
asyncStorage: AsyncStorage,
},
})
);`

It using the "withWalletConnect"

**Reply by ignaciosantise** [reply]
@on99yu you don't need withWalletConnect anymore. 

Try removing that logic and just call `registerRootComponent(App)`

**Reply by on99yu** [reply]
Ok I'll try it
